tests/kernel/threads/thread_apis/src/test_threads_suspend_resume.c

```diff
@@ -27,6 +27,18 @@ static void threads_suspend_resume(int prio)
 	k_tid_t tid = k_thread_create(&tdata, tstack, STACK_SIZE,
 				      thread_entry, NULL, NULL, NULL,
 				      create_prio, K_USER, K_NO_WAIT);
+	/* checkpoint: resume an unsuspended thread has no effect */
+	k_thread_resume(tid);
+	k_msleep(100);
+	zassert_true(last_prio == create_prio, NULL);
```

Comment:
still not sure how this assert on priorities is testing that resume has no effect on unsuspended threads.

Reply:
I added the line 30-34 in this existing function.  The main purpose of these code is to make a comparison between line 42-46. The line 36-37 is just to restore the enviroment.
If I changed the line 31 k_thread_resume to fucntion k_thread_suspend. Then the assert on line 33 would fail.
The new added code can also reflect k_thread_resume() and k_thread_suspend() are totally different function. For example, they are not like mouse buttons, but like light switches.

Reply:
@nashif  Could you take another look at it, thanks!

Reply:
I am still not convinced you are testing what you claim. The priority of the thread does not indicate anything about what happens when you resume a running thread.

Reply:
I got you point.  It have changed the code use kernel API to get thread state to judge zassert. 
I've created a new testcase. The new testcase must test in kernel mode because get thread state is not allowed in user mode.